packages/eslint-plugin-wpcalypso/lib/rules/i18n-unlocalized-url.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file Disallow using unlocalized URL strings
+ * @author Automattic
+ * @copyright 2023 Automattic. All rights reserved.
+ * See LICENSE.md file in root directory for full license.
+ */
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+const LOCALIZABLE_URLS = [
+	'apps.wordpress.com',
+	'automattic.com/cookies(/|$)',
+	'automattic.com/privacy(/|$)',
+	'wordpress.com/blog(/|$)',
+	'wordpress.com/forums(/|$)',
+	'wordpress.com/go(/|$)',
+	'wordpress.com/support(/|$)',
+	'wordpress.com/tos(/|$)',
+];
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+const getCallee = require( '../util/get-callee' );
+
+/**
+ * Check whether the provided string is localizable string.
+ *
+ * @param   {string}  string String to be tested
+ * @returns {boolean}
+ */
+function isLocalizableUrlString( string ) {
+	return LOCALIZABLE_URLS.some( ( url ) =>
+		new RegExp( `^(https?:)?//${ url.replace( '.', '\\.' ) }`, 'i' ).test( string )
+	);
+}
+
+/**
+ * Check whether the provided node type is variable declarator.
+ *
+ * @param   {string}  type Node type
+ * @returns {boolean}
+ */
+function isVariableDeclarationValue( type ) {
+	return type === 'VariableDeclarator';
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const rule = ( module.exports = function ( context ) {
+	const variableDeclarationValues = [];
+	const urlVariables = new Set();
+
+	/**
+	 * Check whether the node is unlocalized url.
+	 *
+	 * @param   {Object} node
+	 * @param   {string} nodeValueString
+	 * @returns {void}
+	 */
+	function handleUnlocalizedUrls( node, nodeValueString ) {
+		// Node is wrapped in localizeUrl, therefore we can assume it's localized and we don't need to do any further checks.
+		if ( getCallee( node.parent ).name === 'localizeUrl' ) {
```

Comment:
Does this work with aliases? There is at least one of those cases in the codebase:

https://github.com/Automattic/wp-calypso/blob/ce985467adb53ece4d878bfcfacbd120c1881ac1/packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx#L3

Reply:
Aliases are currently used during the check. I came across the same use case during the development and thought about covering the `localizeUrl` aliases as well, but decided not to as it seems like a very rare use cases that could just be inline ignored if needed and also for the sake of keeping the implementation not to overcomplicated.

However, iterate on it later if it happens to be a more common case.

Reply:
Sounds good 👍